From: Michael Cree <mcree@orcon.net.nz>
To: linux-alpha@vger.kernel.org
Cc: Michael Cree <mcree@orcon.net.nz>, linux-kernel@vger.kernel.org, Matt
 Turner <mattst88@gmail.com>, Richard Henderson <rth@twiddle.net>, Ivan
 Kokshaysky <ink@jurassic.park.msu.ru>, 649468@bugs.debian.org, Ben
 Hutchings <ben@decadent.org.uk>
Subject: [PATCH] alpha: add io{read,write}{16,32}be functions
Date: Wed, 30 Nov 2011 21:01:40 +1300
Message-Id: <1322640100-14534-1-git-send-email-mcree@orcon.net.nz>

commit 25534eb7707821b796fd84f7115367e02f36aa60 upstream.

These functions are used in some PCI drivers with big-endian
MMIO space.

Admittedly it is almost certain that no one this side of the
Moon would use such a card in an Alpha but it does get us
closer to being able to build allyesconfig or allmodconfig,
and it enables the Debian default generic config to build.

Signed-off-by: Michael Cree <mcree@orcon.net.nz>
---
 arch/alpha/include/asm/io.h |    5 +++++
 1 files changed, 5 insertions(+), 0 deletions(-)

Index: linux-3.2.46/arch/alpha/include/asm/io.h
===================================================================
--- linux-3.2.46.orig/arch/alpha/include/asm/io.h	2013-07-26 18:56:28.000000000 +0000
+++ linux-3.2.46/arch/alpha/include/asm/io.h	2013-07-26 19:17:37.000000000 +0000
@@ -490,6 +490,11 @@
 }
 #endif
 
+#define ioread16be(p) be16_to_cpu(ioread16(p))
+#define ioread32be(p) be32_to_cpu(ioread32(p))
+#define iowrite16be(v,p) iowrite16(cpu_to_be16(v), (p))
+#define iowrite32be(v,p) iowrite32(cpu_to_be32(v), (p))
+
 #define inb_p		inb
 #define inw_p		inw
 #define inl_p		inl
